File: apollo-integration-tests/testFixtures/basic/expected/type/CustomType.java
package type;

import com.apollographql.android.api.graphql.ScalarType;
import java.lang.Class;
import java.lang.Override;
import java.lang.String;
import java.util.Date;
import javax.annotation.Generated;

@Generated("Apollo GraphQL")
public enum CustomType implements ScalarType {
  DATETIME {
    @Override
    public String typeName() {
      return "DateTime";
    }

    @Override
    public Class javaType() {
      return Date.class;
    }
  }
}

File: apollo-compiler/src/test/graphql/com/example/inline_fragment_for_non_optional_field/type/CustomType.kt
package com.example.inline_fragment_for_non_optional_field.type

import com.apollographql.apollo.api.ScalarType
import java.lang.Class
import javax.annotation.Generated
import kotlin.String

@Generated("Apollo GraphQL")
enum class CustomType : ScalarType {
    ID {
        override fun typeName(): String = "ID"

        override fun javaType(): Class<*> = kotlin.String::class.java
    }
}
